AutonomousDemo/Position.cs

```diff
@@ -2,21 +2,24 @@
 
 namespace AutonomousDemo
 {
-    public class Position 
+    public struct Position 
     {
-        public double XCoordinate { get; set; } = 0;
-        public double YCoordinate { get; set; } = 0;
+        public double XCoordinate;
+        public double YCoordinate;
 
-        public Position(double xCoordinate, double yCoordinate)
+        public Position(double xCoordinate = 0, double yCoordinate = 0)
         {
             this.XCoordinate = xCoordinate;
             this.YCoordinate = yCoordinate;
         }
 
-        public Position TravelForOneSecond(Velocity velocity)
+        public Position TravelForOneSecond(Velocity currentVelocity, Position previousPosition)
         {
-            var x =  velocity.Speed * Math.Cos(velocity.Direction);
-            var y = velocity.Speed * Math.Sin(velocity.Direction);
+
+            var _x = currentVelocity.Speed * Math.Cos(currentVelocity.Direction);
```

Comment:
"_x" and "_y" are some confusing variable names. Maybe like "xOffset" or "deltaX"? ('Delta', written as the greek letter, is usually used for 'change in' in math/physics.)